### ISSUE TITLE
Amount formatting issue

### ISSUE THREAD

**Issue by ravinderk** [WordImpress/Give#693]
When thousand and decimal separator set to comma and dot then `give_sanitize_amount` works fine but if we change decimal separator to comma ( same as thousand separator ) then we will get worng goal amount. Since we give option to user to set custom separator then we have to handle this issue. 

<img width="753" alt="screen shot 2016-07-07 at 11 59 48 am" src="https://cloud.githubusercontent.com/assets/1784821/16644403/5b7889be-443a-11e6-859f-48fb3bc6a02e.png">

<img width="418" alt="screen shot 2016-07-07 at 11 59 20 am" src="https://cloud.githubusercontent.com/assets/1784821/16644401/5b5b6cc6-443a-11e6-95ea-0120cd0a46d6.png">

<img width="657" alt="screen shot 2016-07-07 at 11 59 33 am" src="https://cloud.githubusercontent.com/assets/1784821/16644402/5b69fd04-443a-11e6-86ff-99a5da4a99e4.png">

**I have few questions:**
1. Do we need to store formatted goal amount. Cann't we  store this without formatted digit as we are doing for form earnings
<img width="558" alt="screen shot 2016-07-07 at 12 06 53 pm" src="https://cloud.githubusercontent.com/assets/1784821/16644535/5362e4c6-443b-11e6-8286-d9cfecca3e22.png">

2. Current `give_sanitize_amount` function only format amount on basis on current separator's value. Are we handling user case when user will change separator value after some time?

**Comment by ravinderk** [comment]
@DevinWalker I researched about amount formatting issue. and I found that we should not use amount formatting in input fields ( goal and price both ), because we have to remove number formatting from amount whenever we will save them to database. In my opinion number formatting should be one way means whenever we need to display or for any other reason need formatting amount we can calculate it from decimal formatted amount ( we should use dot for decimal when storing value to database ), So we do not need to sanitize formatted amount.


<img width="742" alt="screen shot 2016-07-08 at 6 23 40 pm" src="https://cloud.githubusercontent.com/assets/1784821/16688052/bd9e0290-4539-11e6-93d6-6e998faa0f84.png">
<img width="843" alt="screen shot 2016-07-08 at 6 23 45 pm" src="https://cloud.githubusercontent.com/assets/1784821/16688053/bda757be-4539-11e6-9973-80c40b1025d5.png">
<img width="742" alt="screen shot 2016-07-08 at 6 27 10 pm" src="https://cloud.githubusercontent.com/assets/1784821/16688054/bdb0fbfc-4539-11e6-9b3c-970213a3bb2a.png">

Woocommerce is following same strategy when  showing pre filled price or saving price to database

<img width="575" alt="screen shot 2016-07-08 at 6 39 39 pm" src="https://cloud.githubusercontent.com/assets/1784821/16688351/5db957b0-453b-11e6-9184-7bf5df3b6d50.png">
